Libraries/StyleSheet/precomputeStyle.js

```diff
@@ -126,6 +132,71 @@ function _multiplyTransform(
 }
 
 /**
+ * Takes a transform matrix and an origin, and applies a transformation to
+ * the matrix to change the transform origin. Mutates the passed in matrix.
+ */
+function _applyTransformOrigin(
+  matrix: Array<number>,
+  origin: Object
+): void {
+  origin = _normalizeOrigin(origin);
+  var translate = MatrixMath.createIdentityMatrix();
+  var untranslate = MatrixMath.createIdentityMatrix();
+  MatrixMath.reuseTranslate3dCommand(
+    translate,
+    origin.x,
+    origin.y,
+    origin.z
+  );
+  MatrixMath.reuseTranslate3dCommand(
+    untranslate,
+    -origin.x,
+    -origin.y,
+    -origin.z
+  );
+  MatrixMath.multiplyInto(matrix, translate, matrix);
+  MatrixMath.multiplyInto(matrix, matrix, untranslate);
+}
+
+/**
+ * Accepts the user passed in "transformOrigin" property, which
+ * in the future could be a something other than an object, and
+ * normalizes it to an {x, y, z} point.
+ * 
+ * Right now this function is little more than an optimized
+ * Object.assign, but in the future it could be more complicated.
+ */
+function _normalizeOrigin(input: Object): Object {
+  var output = { x: 0, y: 0, z: 0 };
+  invariant(
+      false,
```

Comment:
I think it is important to have `transform-origin` CSS-compatible in order to avoid future collisions and misunderstandings. 
> By default, a layer's anchorPoint is (0.5, 0.5), which lies at the center of the layer

```objectivec
// Obj-C Fix
view.layer.anchorPoint = CGPointMake(0, 0);
```
Which could be applied somewhere here in [React/Views/RCTViewManager.m](https://github.com/facebook/react-native/blob/a26b89f8bd7d6db66614095afeaf0ce6a1982378/React/Views/RCTViewManager.m#L130)

Or even we could make a separate property and avoid matrix math on JS-side. This also fixes 3-rd issue.
```objectivec
RCT_CUSTOM_VIEW_PROPERTY(transformOrigin, RCTTransformOrigin, RCTView)
{
  view.layer.anchorPoint = json ? [RCTConvert RCTTransformOrigin:json] : CGPointMake(0, 0);
  // or 
  view.layer.anchorPoint = json ? [RCTConvert RCTTransformOrigin:json] : defaultView.layer.anchorPoint;
}

// RCTTransformOrigin and [RCTConvert RCTTransformOrigin] to be defined
// defaultView.layer.anchorPoint to be defined as CGPointMake(0, 0)
```

 - We could use `anchorPoint` instead of multiplying matrixes in order to support `%` values (with the default value of 0, 0).
 - Pixel values we could be casted to `%` using `view.frame`.

@vjeux any ideas how to fix it a better way?